### ISSUE TITLE
ng extract-i18n extracts incorrect `equiv-text` value

### ISSUE THREAD

**Issue by kamilchlebek** [angular/angular#39671]
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/localize

### Is this a regression?

Yes, the previous version in which this bug was not present was: v10

### Description

Component template (it's usually formatted like this by prettier):
```
<div i18n>
  test {{ title }}
</div>
```

Running `ng xi18n` in angular v10, produced output that contained `source` element that looked:
```
<source>
   test <x id="INTERPOLATION" equiv-text="{{ title }}"/>
</source> 
```

Now after running `ng extract-i18n` it's:
```
<source> test <x id="INTERPOLATION" equiv-text="t {{ title "/>
</source>
```

## 🔬 Minimal Reproduction

https://github.com/kamilchlebek/equiv-text-angular11 (see `messages.xlf`)

## 🌍  Your Environment 

(it's a brand new project created using `ng new`)

**Angular Version:**
<pre><code>Angular CLI: 11.0.1
Node: 12.16.1
OS: linux x64

Angular: 11.0.0
... animations, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1100.1
@angular-devkit/build-angular   0.1100.1
@angular-devkit/core            11.0.1
@angular-devkit/schematics      11.0.1
@angular/cli                    11.0.1
@schematics/angular             11.0.1
@schematics/update              0.1100.1
rxjs                            6.6.3
typescript                      4.0.5
</code></pre>

**Comment by petebacondarwin** [comment]
I can confirm this bug. I am looking into it now.

**Comment by petebacondarwin** [comment]
Just to give an update... these `equiv-text` properties are computed via the source-map from the generated code (e.g. `app.component.js`) back to the original source (e.g. `app.component.html`). In this case the source-map that was generated for this file is not accurate. See this [visualization](https://sokra.github.io/source-map-visualization/#base64,aW1wb3J0IHsgQ29tcG9uZW50IH0gZnJvbSAnQGFuZ3VsYXIvY29yZSc7CmltcG9ydCAqIGFzIGkwIGZyb20gIkBhbmd1bGFyL2NvcmUiOwpleHBvcnQgY2xhc3MgQXBwQ29tcG9uZW50IHsKICAgIGNvbnN0cnVjdG9yKCkgewogICAgICAgIHRoaXMudGl0bGUgPSAnZXF1aXYtdGV4dCc7CiAgICB9Cn0KQXBwQ29tcG9uZW50LsOJwrVmYWMgPSBmdW5jdGlvbiBBcHBDb21wb25lbnRfRmFjdG9yeSh0KSB7IHJldHVybiBuZXcgKHQgfHwgQXBwQ29tcG9uZW50KSgpOyB9OwpBcHBDb21wb25lbnQuw4nCtWNtcCA9IGkwLsOJwrXDicK1ZGVmaW5lQ29tcG9uZW50KHsgdHlwZTogQXBwQ29tcG9uZW50LCBzZWxlY3RvcnM6IFtbImFwcC1yb290Il1dLCBkZWNsczogNCwgdmFyczogMiwgY29uc3RzOiBmdW5jdGlvbiAoKSB7IGxldCBpMThuXzA7IGlmICh0eXBlb2YgbmdJMThuQ2xvc3VyZU1vZGUgIT09ICJ1bmRlZmluZWQiICYmIG5nSTE4bkNsb3N1cmVNb2RlKSB7CiAgICAgICAgY29uc3QgTVNHX0VYVEVSTkFMXzIzODgzNzA2OTEyNDAwMzUwMTUkJF9fX19fX19fX19fX19fX19fX1VTRVJTX1BFVEVfREVWX0FOR1VMQVJfVE1QX0lTU1VFXzM5NjcxX1NSQ19BUFBfQVBQX0NPTVBPTkVOVF9UU18xID0gZ29vZy5nZXRNc2coIiB0ZXN0IHskaW50ZXJwb2xhdGlvbn1cbiIsIHsgImludGVycG9sYXRpb24iOiAiXHVGRkZEMFx1RkZGRCIgfSk7CiAgICAgICAgaTE4bl8wID0gTVNHX0VYVEVSTkFMXzIzODgzNzA2OTEyNDAwMzUwMTUkJF9fX19fX19fX19fX19fX19fX1VTRVJTX1BFVEVfREVWX0FOR1VMQVJfVE1QX0lTU1VFXzM5NjcxX1NSQ19BUFBfQVBQX0NPTVBPTkVOVF9UU18xOwogICAgfQogICAgZWxzZSB7CiAgICAgICAgaTE4bl8wID0gJGxvY2FsaXplIGA6w6LCkMKfNmZlMTllZmE3MzIzYzQyYzUyNDlhODdhYTkyZGQ1Y2U5YzVkODdjZcOiwpDCnzIzODgzNzA2OTEyNDAwMzUwMTU6IHRlc3QgJHsiXHVGRkZEMFx1RkZGRCJ9OklOVEVSUE9MQVRJT046CmA7CiAgICB9IGxldCBpMThuXzI7IGlmICh0eXBlb2YgbmdJMThuQ2xvc3VyZU1vZGUgIT09ICJ1bmRlZmluZWQiICYmIG5nSTE4bkNsb3N1cmVNb2RlKSB7CiAgICAgICAgY29uc3QgTVNHX0VYVEVSTkFMXzMyOTcyODY3NTQ1OTYwNjgyOTkkJF9fX19fX19fX19fX19fX19fX1VTRVJTX1BFVEVfREVWX0FOR1VMQVJfVE1QX0lTU1VFXzM5NjcxX1NSQ19BUFBfQVBQX0NPTVBPTkVOVF9UU18zID0gZ29vZy5nZXRNc2coInRlc3QyIHskaW50ZXJwb2xhdGlvbn0iLCB7ICJpbnRlcnBvbGF0aW9uIjogIlx1RkZGRDBcdUZGRkQiIH0pOwogICAgICAgIGkxOG5fMiA9IE1TR19FWFRFUk5BTF8zMjk3Mjg2NzU0NTk2MDY4Mjk5JCRfX19fX19fX19fX19fX19fX19VU0VSU19QRVRFX0RFVl9BTkdVTEFSX1RNUF9JU1NVRV8zOTY3MV9TUkNfQVBQX0FQUF9DT01QT05FTlRfVFNfMzsKICAgIH0KICAgIGVsc2UgewogICAgICAgIGkxOG5fMiA9ICRsb2NhbGl6ZSBgOsOiwpDCnzE1MjcxODA4ODM3NjFjMDJlYjM1NzE2MTk1YWZhNWM2M2YxZGY5MjPDosKQwp8zMjk3Mjg2NzU0NTk2MDY4Mjk5OnRlc3QyICR7Ilx1RkZGRDBcdUZGRkQifTpJTlRFUlBPTEFUSU9OOmA7CiAgICB9IHJldHVybiBbaTE4bl8wLCBpMThuXzJdOyB9LCB0ZW1wbGF0ZTogZnVuY3Rpb24gQXBwQ29tcG9uZW50X1RlbXBsYXRlKHJmLCBjdHgpIHsgaWYgKHJmICYgMSkgewogICAgICAgIGkwLsOJwrXDicK1ZWxlbWVudFN0YXJ0KDAsICJkaXYiKTsKICAgICAgICBpMC7DicK1w4nCtWkxOG4oMSwgMCk7CiAgICAgICAgaTAuw4nCtcOJwrVlbGVtZW50RW5kKCk7CiAgICAgICAgaTAuw4nCtcOJwrVlbGVtZW50U3RhcnQoMiwgImRpdiIpOwogICAgICAgIGkwLsOJwrXDicK1aTE4bigzLCAxKTsKICAgICAgICBpMC7DicK1w4nCtWVsZW1lbnRFbmQoKTsKICAgIH0gaWYgKHJmICYgMikgewogICAgICAgIGkwLsOJwrXDicK1YWR2YW5jZSgxKTsKICAgICAgICBpMC7DicK1w4nCtWkxOG5FeHAoY3R4LnRpdGxlKTsKICAgICAgICBpMC7DicK1w4nCtWkxOG5BcHBseSgxKTsKICAgICAgICBpMC7DicK1w4nCtWFkdmFuY2UoMik7CiAgICAgICAgaTAuw4nCtcOJwrVpMThuRXhwKGN0eC50aXRsZSk7CiAgICAgICAgaTAuw4nCtcOJwrVpMThuQXBwbHkoMyk7CiAgICB9IH0sIHN0eWxlczogWyIiXSB9KTsKLypAX19QVVJFX18qLyAoZnVuY3Rpb24gKCkgeyBpMC7DicK1c2V0Q2xhc3NNZXRhZGF0YShBcHBDb21wb25lbnQsIFt7CiAgICAgICAgdHlwZTogQ29tcG9uZW50LAogICAgICAgIGFyZ3M6IFt7CiAgICAgICAgICAgICAgICBzZWxlY3RvcjogJ2FwcC1yb290JywKICAgICAgICAgICAgICAgIHRlbXBsYXRlVXJsOiAnLi9hcHAuY29tcG9uZW50Lmh0bWwnLAogICAgICAgICAgICAgICAgc3R5bGVVcmxzOiBbJy4vYXBwLmNvbXBvbmVudC5jc3MnXQogICAgICAgICAgICB9XQogICAgfV0sIG51bGwsIG51bGwpOyB9KSgpOwovLyMgc291cmNlTWFwcGluZ1VSTD1hcHAuY29tcG9uZW50LmpzLm1hcA==,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiYXBwLmNvbXBvbmVudC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIi4uLy4uLy4uL3NyYy9hcHAvYXBwLmNvbXBvbmVudC50cyIsIi4uLy4uLy4uL3NyYy9hcHAvYXBwLmNvbXBvbmVudC5odG1sIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBLE9BQU8sRUFBRSxTQUFTLEVBQUUsTUFBTSxlQUFlLENBQUM7O0FBTzFDLE1BQU0sT0FBTyxZQUFZO0lBTHpCO1FBTUUsVUFBSyxHQUFHLFlBQVksQ0FBQztLQUN0Qjs7d0VBRlksWUFBWTtpREFBWixZQUFZOzs7OztpQkNOdkIsVUFEUSx3RUFDTCxlQUFXO0NBQUMsQUFDakI7Ozs7OztpQkFDVSxVQUFBLHdFQUFNLGVBQVcsaUJBQUE7O1FBSDNCLDJCQUNFO1FBREYsZUFDRTtRQUNGLGlCQUFNO1FBQ04sMkJBQVU7UUFBVixlQUFVO1FBQWlCLGlCQUFNOztRQURqQyxlQUFNO1FBQU4sdUJBQU07UUFBTixpQkFBTTtRQUNxQixlQUFNO1FBQU4sdUJBQU07UUFBTixpQkFBTTs7a0RESXBCLFlBQVk7Y0FMeEIsU0FBUztlQUFDO2dCQUNULFFBQVEsRUFBRSxVQUFVO2dCQUNwQixXQUFXLEVBQUUsc0JBQXNCO2dCQUNuQyxTQUFTLEVBQUUsQ0FBQyxxQkFBcUIsQ0FBQzthQUNuQyJ9,aW1wb3J0IHsgQ29tcG9uZW50IH0gZnJvbSAnQGFuZ3VsYXIvY29yZSc7CgpAQ29tcG9uZW50KHsKICBzZWxlY3RvcjogJ2FwcC1yb290JywKICB0ZW1wbGF0ZVVybDogJy4vYXBwLmNvbXBvbmVudC5odG1sJywKICBzdHlsZVVybHM6IFsnLi9hcHAuY29tcG9uZW50LmNzcyddCn0pCmV4cG9ydCBjbGFzcyBBcHBDb21wb25lbnQgewogIHRpdGxlID0gJ2VxdWl2LXRleHQnOwp9Cg==,PGRpdiBpMThuPgogIHRlc3Qge3sgdGl0bGUgfX0KPC9kaXY+CjxkaXYgaTE4bj50ZXN0MiB7eyB0aXRsZSB9fTwvZGl2Pgo=).

I think this is a bug in the Angular compiler, similar to one that I fixed recently. See https://github.com/angular/angular/pull/39486.

Investigating further...

**Comment by petebacondarwin** [comment]
It seems that the new lines in the message are what is messing things up. If you change the template to:

```html
<div i18n>  test {{ title }}
</div>
```

Then the output is accurate.

If you add even more newlines in then it gets worse:

```
<div i18n>
  
  test {{ title }}
</div>
```

resulting in:

```
equiv-text="test {{ tit"
```

**Comment by kamilchlebek** [comment]
That's bad :disappointed: 

As I workaround I could switch prettier `htmlWhitespaceSensitivity` to `strict`. Then the markup will look ugly:
```
<div i18n
  >test {{ title }}</div
>
```

but `messages.xlf` are okay. 

Of course I'd rather avoid that for `div` and other elements being `display:block` by default. 

https://prettier.io/blog/2018/11/07/1.15.0.html#whitespace-sensitive-formatting

**Comment by petebacondarwin** [comment]
I have found the cause of the problem. When we process the `i18n` tags, we run the processing twice. Once before whitespace has been removed (which results in correct source-spans) and then again after whitespace has been collapsed (which results in incorrect source-spans for placeholders).

See https://github.com/angular/angular/blob/1f956184c426814be8fcbdab03396c910e29d935/packages/compiler/src/render3/view/template.ts#L2085-L2088

A fix is in the works...
